src/components/network-chart/NetworkChartRow.js

```diff
@@ -116,6 +129,63 @@ class NetworkChartRowBar extends React.PureComponent<NetworkChartRowBarProps> {
     return markerPosition;
   }
 
+  _preconnectComponent(): React.Node {
+    const { networkPayload, marker } = this.props;
+
+    const preconnectStart = networkPayload.domainLookupStart;
+    if (typeof preconnectStart !== 'number') {
+      // All preconnect operations include a domain lookup part.
+      return null;
+    }
+
+    // The preconnect bar goes from the start to the end of the whole preconnect
+    // operation, that includes both the domain lookup and the connection
+    // process. Therefore we want the property that represents the latest phase.
+    const latestPreconnectEndProperty = PRECONNECT_END_PROPERTIES.find(
```

Comment:
I find this section to be a bit hard to follow, could it be simplified to:

```
let preconnectEnd;
if (typeof networkPayload.connectEnd === 'number') {
  preconnectEnd = networkPayload.connectEnd;
} else if (typeof networkPayload.domainLookupEnd === 'number') {
  preconnectEnd = networkPayload.domainLookupEnd;
} else {
  return null;
}
```

This would make it all self-contained, and relatively easy to follow-along with. I'm not sure if Flow will want a `'connectEnd' in networkPayload` check as well. I'd even prefer coercing the object into something where it's easy to extract values.

Reply:
When I wrote this logic I didn't know how many properties I'd need to loop through. But with only 2 properties I think it's better your way indeed. I'll change it in the chart as well (we have the same logic there).
Nope we don't need the `in` check, actually Flow doesn't use the `in` checks to refine types and `typeof` checks is the best way to do it :)